### PR TITLE
add mkdocs pages

### PR DESCRIPTION
**Pull request by mirpedrol**

### CODE REVIEW

File: docs/contributing/setup.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+```bash
+./gradlew check
+```
+
+## Launch without a local Nextflow build
+
+The plugin can be tested prior release without using a local Nextflow build using the following steps:
+
+1. Build the plugin: `make buildPlugins`
```
Comment (mashehu) [inline docs/contributing/setup.md:20]:
```suggestion
1. Build the plugin: 
```bash
make buildPlugins
```
```

Comment (mashehu) [reply docs/contributing/setup.md:20]: with this you get the copy code button

File: docs/contributing/setup.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+To test with Nextflow for development purpose:
+
+1. Clone the Nextflow repo into a sibling directory
+
+   ```bash
+   cd .. && https://github.com/nextflow-io/nextflow
+   cd nextflow && ./gradlew exportClasspath
+   ```
+
+2. Append to the `settings.gradle` in this project the following line:
```
Comment (mashehu) [inline docs/contributing/setup.md:39]:
```suggestion
2. Append the following line to the `settings.gradle` in this project:
```

File: docs/contributing/setup.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+   includeBuild('../nextflow')
+   ```
+
+3. Compile the plugin code
+
+   ```bash
+   ./gradlew compileGroovy
+   ```
+
+4. run nextflow with this command:
```
Comment (mashehu) [inline docs/contributing/setup.md:51]:
```suggestion
4. Run nextflow with this command:
```

File: docs/index.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
+---
+title: nf-co2footprint
+description: Nextflow plugin for CO2 footprint estimations
```
Comment (mashehu) [inline docs/index.md:3]:
```suggestion
description: Nextflow plugin for CO<sup>2</sup> footprint estimations
```

Comment (mashehu) [reply docs/index.md:3]: not sure if this works

File: docs/usage/parameters.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
+---
+title: Parameters
+description: Customising parameters for the CO2e calculation.
```
Comment (mashehu) [inline docs/usage/parameters.md:3]:
```suggestion
description: Customising parameters for the CO<sup>2</sup>e calculation.
```

File: mkdocs.yml [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    - co2footprint.md
+  - Usage:
+      - usage/parameters.md
+  - Contributing:
+      - contributing/setup.md
+
+theme:
+  name: material
+  icon:
+    logo: octicons/checklist-16
```
Comment (mashehu) [inline mkdocs.yml:19]:
```suggestion
    logo: material/tree
```

**Review by mashehu — APPROVED** [review]

File: docs/index.md (hunk 4 above)
Comment (mirpedrol) [reply docs/index.md:3]: Isn't it better to have plain text in the description?

File: docs/co2footprint.md [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+$C = t * (n_c * P_c * u_c + n_m * P_m) * PUE * CI$
+
+Being:
+
+- **$t$** the running time of the computation (hours) 
+- **$n_c$** the number of cores
+- **$n_m$** the size of memory available (gigabytes)
+- **$u_c$** the core usage factor (between 0 and 1)
+- **$P_c$** the power draw of a computing core 
+- **$P_m$** the power draw of memory (Watt)
```
Comment (skrakau) [inline docs/co2footprint.md:25]:
```suggestion
- **$P_m$** the power draw of memory (Watt, per GB)
```

File: docs/co2footprint.md [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+The equation used for the calculation of the carbon footprint ($C$) is:
+
+$C = t * (n_c * P_c * u_c + n_m * P_m) * PUE * CI$
+
+Being:
+
+- **$t$** the running time of the computation (hours) 
+- **$n_c$** the number of cores
+- **$n_m$** the size of memory available (gigabytes)
```
Comment (skrakau) [inline docs/co2footprint.md:22]:
```suggestion
- **$n_m$** the size of memory available (GB)
```
maybe just GB?

File: docs/co2footprint.md [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+- **$u_c$** the core usage factor (between 0 and 1)
+- **$P_c$** the power draw of a computing core
```
Comment (skrakau) [inline docs/co2footprint.md:24]:
```suggestion
- **$P_c$** the power draw of a computing core (W)
```

**Review by skrakau — APPROVED** [review]
Looks good!

**Pull request merged**